stdlib/bytearray.mini

```diff
@@ -129,165 +126,139 @@ public func bytearray_size(ba: ByteArray) -> uint {
+            return asm(8*(32-bytesToRead), mask & getbuffer256(ba.contents, offset+bytesToRead-32)) uint { shl };
+        } else {
+            return getbuffer256(ba.contents, offset);
         }
-        return ret;
     }
+    offset = offset + ba.sliceOffset;
+    let res = getbuffer256(ba.contents, offset);
+    return res;
+}
+
+func make(cont: ByteArray) -> ByteArray {
+    return struct {
+        size: cont.size + cont.sliceOffset + 32,
+        sliceOffset: 0,
+        contents: cont.contents,
+        copy: true,
+    };
+}
+
+public func bytearray_wrap(cont: buffer, size: uint) -> ByteArray {
+    return struct {
+        size: size,
+        sliceOffset: 0,
+        contents: cont,
+        copy: false,
+    };
 }
 
 // bytearray_setbyte writes one byte to a ByteArray, returns the resulting ByteArray
-public func bytearray_setByte(ba: ByteArray, offset: uint, val: uint) -> ByteArray {
-    if (offset >= ba.size) {
-        if (offset >= 0x100000000000000000000000000000000) {
-            return ba;    // ignore out-of-bounds write
-        }
-        ba = ba with { size: offset+1 };
+public func bytearray_setByte(ba: ByteArray, offset: uint, value: uint) -> ByteArray {
+    if (offset >= const::MaxByteArraySize) {
+        return ba;    // ignore out-of-bounds write
     }
-    offset = offset + ba.sliceOffset;
-    let (newContents, _) = expandingIntArray_op(
-        ba.contents, 
-        offset/32, 
-        struct {
-            f: unsafecast<OpClosureFunc>(basb_closure_func),
-            val: (31-(offset%32), val,),
+    if (offset >= const::MaxByteArraySize) {
+        ba = bytearray_copy(ba, 0, bytearray_new(0), 0, ba.size);
+        if (offset+1 > ba.size) {
+            ba = ba with { size: offset+1 };
+        }
+    } else {
+        if (offset+1 > ba.size) {
+            ba = ba with { size: offset+1 };
         }
-    );
-    return ba with { contents: newContents };
+        offset = offset + ba.sliceOffset;
+    }
+    return ba with { contents: setbuffer8(ba.contents, offset, value) };
 }
 
 // bytearray_set64 writes an 8-byte chunk to a ByteArray, returning the resulting ByteArray
 public func bytearray_set64(ba: ByteArray, offset: uint, value: uint) -> ByteArray {
-    if (offset >= 0x100000000000000000000000000000000-7) {
+    if (offset >= const::MaxByteArraySize - 7) {
         return ba;    // ignore out-of-bounds write
     }
-    if (offset+8 > ba.size) {
-        ba = ba with { size: offset+8 };
-    }
-    offset = offset + ba.sliceOffset;
-    let lowWordIndex = offset / 32;
-    let alignment = offset % 32;
-    if (alignment <= 24) {
-        let (newContents, _) = expandingIntArray_op(
-            ba.contents,
-            lowWordIndex,
-            struct{
-                f: unsafecast<OpClosureFunc>(bas64_single_closure_func),
-                val: (alignment, value,),
-            }
-        );
-        return ba with { contents: newContents };
+    if (offset >= const::MaxByteArraySize - 7) {
```

Comment:
This conditional repeats the first and should be corrected

Reply:
Fixed.